### PR TITLE
StartConfiguration changes

### PR DESCRIPTION
**Pull request by Ryplinn**
I discovered Nightgamesmod last week and played version 2.0.3.3 for a few days straight. After trying out a bunch of player builds, I decided to delve into NewGame+. However, I noticed two major problems with all advanced starts: 1) The game was ignoring the way I spent my attribute points on the starting screen, and 2) The NPCs were missing a lot of their usual aspects. Poor Angel lost her glorious boobs and cute outfit!

Long story short, I made a bunch of changes to game creation, character creation, and start configuration management. The philosophy I took with regards to handling start configs is that they represent modifications to the base game, that anything not specified in the config should be left alone (as much as it makes sense to do so), and that if you find yourself trying to completely rewrite a character in a config you should probably just make a custom NPC instead.

If you like what I've done here, I have a lot of free time until July and would be happy to work on any bug fixes or new features you have in mind.

TL;DR: Fixed (maybe) advanced starts, made game and character creation more consistent, and added some rudimentary unit tests.

### CODE REVIEW

**Comment by dndw** [comment]
Thanks for this. I'll have a look later and then merge it in.
